### ISSUE TITLE
LSC Smart Mood Light (3004154) - fully working

### ISSUE THREAD

**Issue by maarten682** [openshwprojects/OpenBK7231T_App#207]
Just pulled one of these lights open (its BK7231N based), soldered some wires and flashed this firmware and i can happily report that it pretty much works out of the box. Here are the pictures i made, shout if you need better quality or anything else to get this officially supported in your project;

https://i.imgur.com/4wbDSwr.jpg
https://i.imgur.com/VAM7muo.jpg
https://i.imgur.com/tTq0g7W.jpg
https://i.imgur.com/wv8VJiU.jpg
https://i.imgur.com/kR7M7fj.png
https://i.imgur.com/wiN5oKA.png

**Comment by openshwprojects** [comment]
Very nice photos, could you do the honors and post them in our teardowns section here:
https://www.elektroda.com/rtvforum/forum507.html
with a short information where the device was bought and how was configured?

I will add later this to the online database (devices.json) of webapp.

**Comment by beaune33** [comment]
Perfect! One question: Did you try to flash the device without opening using tuya-cloud-cutter? For older versions of this device this should work, for newer ones there is an issue: https://github.com/tuya-cloudcutter/tuya-cloudcutter/issues/125. To fix it they need a firmware dump, which can only be achieved by onening the device, as you already did it. Unfortunately it is probably too late now for the dump because openbeken is already flashed. But maybe you could do if you open another device. by the way, did you manage to open it without destroying the housing? How?

**Comment by beaune33** [comment]
One more thing: he device has got a pushbutton. Did you also manage to get that working with openbeken?

**Comment by maarten682** [comment]
> Very nice photos, could you do the honors and post them in our teardowns section here: https://www.elektroda.com/rtvforum/forum507.html with a short information where the device was bought and how was configured?
> 
> I will add later this to the online database (devices.json) of webapp.

Sure! Also, for the full configuration the momentary pushbutton on the bottom of the device is connected to P14.

**Comment by maarten682** [comment]
> Did you try to flash the device without opening

I did not, it might be possible but i already hooked the device up with the app so i assumed that would have updated and patched the firmware so i honestly did not bother. And i prefer the simplicity and reliability of hardware flashing over jumping through weird hoops with ota stuff so im not very tempted to try on my next devices. These are so incredibly easy to open that i dont see a real advantage (other than not having to solder).

**Comment by maarten682** [comment]
> One more thing: he device has got a pushbutton. Did you also manage to get that working with openbeken?

Yes, did not try it too in-depth but the button is connected to P14 and i just configured it to toggle the warm white so i can grab the device and use it as a flashlight when i need to see something behind my tv ;)

https://i.imgur.com/spXf4cT.png

Little word of warning, doing it like this will turn on the CW at full power without turning off any of the other leds so if you already had a fairly power hungry combination of RGB going the total power draw might overheat you little light. Use with caution. There are probably better ways to do it but this serves my purpose.
